### PR TITLE
Add ability to synchronize multiple shimmers

### PR DESCRIPTION
**Pull request by arigreen**

### CODE REVIEW

File: FBShimmering/FBShimmeringLayer.m [hunk 1]
```diff
@@ -244,6 +244,22 @@ - (void)setShimmeringOpacity:(CGFloat)shimmeringOpacity
...
+
+- (void)setShimmeringBeginTime:(CFTimeInterval)beginTime
+{
+  CAAnimation *slideAnimation = [[_maskLayer animationForKey:kFBShimmerSlideAnimationKey] mutableCopy];
+  if (nil == slideAnimation) {
+    return;
+  }
+
+  slideAnimation.beginTime = beginTime;
+  [_maskLayer addAnimation:slideAnimation forKey:kFBShimmerSlideAnimationKey];
```
Comment (grp) [inline FBShimmering/FBShimmeringLayer.m]: Rather than adding this here, set an ivar and read that out inside the shimmer update function?

Comment (arigreen) [inline FBShimmering/FBShimmeringLayer.m]: I'm not clear about where in _updateShimmering to set the begin time.

File: FBShimmering/FBShimmeringLayer.m [hunk 2]
```diff
@@ -47,6 +47,8 @@ static void FBShimmeringLayerAnimationApplyDragCoefficient(CAAnimation *animatio
 static NSString *const kFBFadeAnimationKey = @"fade";
 static NSString *const kFBEndFadeAnimationKey = @"fade-end";
 
+static const float FBShimmerDefaultBeginTime = CGFLOAT_MAX;
```
Comment (grp) [inline FBShimmering/FBShimmeringLayer.m]: Maybe expose this?

File: FBShimmering/FBShimmeringLayer.m [hunk 3]
```diff
@@ -442,7 +453,11 @@ - (void)_updateShimmering
       slideAnimation = shimmer_slide_animation(self, animationDuration, _shimmeringDirection);
       slideAnimation.fillMode = kCAFillModeForwards;
       slideAnimation.removedOnCompletion = NO;
-      slideAnimation.beginTime = CACurrentMediaTime() + fadeOutAnimation.duration;
+      if (_shimmeringBeginTime != FBShimmerDefaultBeginTime) {
+        slideAnimation.beginTime = _shimmeringBeginTime;
+      } else {
+        slideAnimation.beginTime = CACurrentMediaTime() + fadeOutAnimation.duration;
```
Comment (grp) [inline FBShimmering/FBShimmeringLayer.m]: only comment: this should set _shimmeringBeginTime to the same value, so you can read it out from the shimmering layer

**Comment by eikebartels** [comment]
Do you have an example how to synchronize multiple shimmers?

**Comment by grp** [comment]
Generally, you'd just set the `shimmeringBeginTime` to the same value on all your shimmering views.

**Comment by eikebartels** [comment]
Ah yea make sense. Thanks a lot
On Wed, 8 Jun 2016 at 19:21, Grant Paul <notifications@github.com> wrote:

> Generally, you'd just set the shimmeringBeginTime to the same value on
> all your shimmering views.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/facebook/Shimmer/pull/47#issuecomment-224664563>, or mute
> the thread
> <https://github.com/notifications/unsubscribe/AAu368TtHV9oFdgji3y4sVmoykHsjJ7Kks5qJvoagaJpZM4ES90Z>
> .
>
-- 
Eike Bartels
Software Engineer
+4915754449623

**Pull request merged**